Userland/Applications/Run/RunWindow.cpp

```diff
@@ -130,6 +128,9 @@ bool RunWindow::run_as_command(ByteString const& run_input)
 
     // 127 is typically the shell indicating command not found. 126 for all other errors.
     if (child_error == 126 || child_error == 127) {
+        remove_all_matching_from_history(run_input);
+        // FIXME: Handle failure to save history somehow.
+        (void)save_history();
```

Comment:
Thanks for the review. Just wish to clarify something:

>  just because its execution failed is usually not part of "proper" history handling.

I remove an entry in case of failure here because Windows 11's `Run`  would do something similar (I couldn't keep the history results to stay open, but the history list is empty) :

![Screenshot 2024-04-05 091529](https://github.com/SerenityOS/serenity/assets/60799661/8aca8a13-ceda-4567-9dab-6895b49b54a7)

I'm not sure how to address it in some other way at the moment, so if you wish I can leave the entry as-is, even in case of failure.

Reply:
:thinking: I'm thinking about this from a shell pov, I suppose it's alright to remove the entry if that's the expected behaviour.

then just mention the change (maybe plus something like "this is more in line with how other impls do it"); otherwise lgtm.

Reply:
Sorry it took so long to make the necessary changes 🙏! At the time I had little access to the computer I use to run the OS.

So long story short, Windows 11's Run doesn't bother removing a program that failed during its runtime. I wrongly assumed at the time that they did, which is why I removed the history entry here. But after passing a path to a python script command line that I made to intentionally crash, Windows still saves this command in the history. So I decided to remove this change.